cluster.go

```diff
@@ -142,88 +179,55 @@ func (c *ClusterClient) process(cmd Cmder) {
 	}
 }
 
-// Closes all connections and reloads slot cache, if due
+// Closes all connections and reloads slot cache, if due.
 func (c *ClusterClient) reloadIfDue() (err error) {
 	if !atomic.CompareAndSwapUint32(&c._reload, 1, 0) {
 		return
 	}
 
-	var infos []ClusterSlotInfo
-
 	c.slotsMx.Lock()
 	defer c.slotsMx.Unlock()
 
-	// Try known addresses in random order (map interation order is random in Go)
-	// http://redis.io/topics/cluster-spec#clients-first-connection-and-handling-of-redirections
-	// https://github.com/antirez/redis-rb-cluster/blob/fd931ed/cluster.rb#L157
-	for addr := range c.addrs {
-		c.reset()
-
-		infos, err = c.fetchClusterSlots(addr)
-		if err == nil {
-			c.update(infos)
-			break
-		}
+	client := c.randomClient()
```

Comment:
I followed antirez' reference implementation here, which suggests to iterate over all known cluster addresses in random order and reset before each step. What if you have no more (working) clients left? It will panic.